src/H5Dchunk.c

```diff
@@ -7711,21 +7711,16 @@ H5D__chunk_iter_cb(const H5D_chunk_rec_t *chunk_rec, void *udata)
 herr_t
 H5D__chunk_iter(const H5D_t *dset, H5D_chunk_iter_op_t op, void *op_data)
 {
-    const H5O_layout_t *layout = NULL;       /* Dataset layout */
-    const H5D_rdcc_t *  rdcc   = NULL;       /* Raw data chunk cache */
-    H5D_rdcc_ent_t *    ent;                 /* Cache entry index */
-    H5D_chk_idx_info_t  idx_info;            /* Chunked index info */
-    herr_t              ret_value = SUCCEED; /* Return value */
+    const H5D_shared_t *const shared = dset->shared;
+    const H5O_layout_t *const layout = &shared->layout;
+    const H5D_rdcc_t *const   rdcc   = &shared->cache.chunk;
+    H5D_rdcc_ent_t *          ent;                 /* Cache entry index */
+    H5D_chk_idx_info_t        idx_info;            /* Chunked index info */
+    herr_t                    ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_PACKAGE_TAG(dset->oloc.addr)
 
-    /* Check args */
-    HDassert(dset);
-    HDassert(dset->shared);
```

Comment:
This is fundamentally introducing a new paradigm of parameter handling. Instead of checking parameters via asserts before assigning to things like dset->shared, the asserts get tossed in favor of direct assignments at the top.

The problem with this is that now there are two paradigms for input parameter handling used in the library and it's not clear which would be the "right" one.

Reply:
Can the asserts be moved to the top of the function before the assignments?

Reply:
I have taken the change out because it is not integral to the PR and I really want the PR merged, but there is no reasonable defense for these assertions.